app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt

```diff
@@ -0,0 +1,356 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@file:Suppress("TooManyFunctions")
+
+package org.mozilla.fenix.tabstray.syncedtabs
+
+import androidx.annotation.VisibleForTesting
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.IntrinsicSize
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Divider
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.geometry.CornerRadius
+import androidx.compose.ui.graphics.PathEffect
+import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import mozilla.components.browser.storage.sync.SyncedDeviceTabs
+import mozilla.components.browser.storage.sync.TabEntry
+import mozilla.components.browser.toolbar.MAX_URI_LENGTH
+import mozilla.components.feature.syncedtabs.view.SyncedTabsView
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.compose.SecondaryText
+import org.mozilla.fenix.theme.FirefoxTheme
+import mozilla.components.browser.storage.sync.Tab as SyncTab
+
+/**
+ * Top-level list UI for displaying Synced Tabs in the Tabs Tray.
+ *
+ * @param syncedTabs The tab UI items to be displayed.
+ * @param listener to apply to this layout node.
+ */
+@Composable
+fun SyncedTabsList(syncedTabs: List<SyncedTabsListItem>, listener: SyncedTabsView.Listener) {
```

Comment:
```suggestion
fun SyncedTabsList(syncedTabs: List<SyncedTabsListItem>, onTabClicked: (Tab) -> Unit) {
```

If you look at how SyncedTabsAdapter is constructed, the TabClickDelegate is used as a weird pass-through to satisfy the API needs for the adapter.

You no longer have this problem in Compose. You can just call the interactor directly!

```kotlin
SyncedTabsList(listOfSyncedTabs) { tab ->
  navigationInteractor.onSyncedTabClicked(tab)
}
```

In the `SyncedTabsTrayLayout` we don't even complete the other end of the call because it's not worth the gymnastics needed to get the listener to work just to have it provide the same data to us in the feature. See this comment [here](https://github.com/mozilla-mobile/fenix/blob/0c34bd93b83ac929ae6b14c3216e95cf7092b79c/app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabsTrayLayout.kt#L53-L57).

The only value the listener now provides is to tell the feature to refresh the list, and you have that covered in `SyncedTabsIntegration` with the [`SyncButtonBinding`](https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncButtonBinding.kt).

What this means, is that it simplifies the refactor in AC because it just needs to be a method on the feature and not by a reference to a listener through a view.

After removing the listener from the feature your `SyncedTabsIntegration` call to refresh would look like this:
```kotlin
private val syncButtonBinding by lazy {
  SyncButtonBinding(store) { syncedTabsFeature.syncNow() }
}
```